08- fraction_calculator/FracCalc.java

```diff
@@ -32,50 +32,48 @@ public static void main(String[] args){
      *      Example: return ==> "1_1/4"
      */
     public static String produceAnswer(String input){
+      if (input.contains("stop") == true){
+        System.out.println("stop");
+        return "stop";
+      }
+
       int end = input.length();
 
       int space = input.indexOf(" ") + 1;
-      //System.out.println(input.indexOf(" ") + 1);
-      String num1 = input.substring(0,space);//get first number
-      //System.out.println(num1);
+    //  System.out.println(input.indexOf(" ") + 1);
+      String num1 = input.substring(0,space-1);//get first number
+      //System.out.println(num1.length());
 
       int space2 = input.indexOf(" ",1) + 3;
-      //System.out.println(input.indexOf(" ") + 3);
+    //  System.out.println(input.indexOf(" ") + 3);
       String num2 = input.substring(space2,end);//second number
 
       String Operation = input.substring(space,space + 1);//locate substring of the operation
-      //System.out.println(Operation);
-      int type;
-// see which kind of operation it is
-      if (Operation.equals("+")){
-        type = 1;
-      }
-      else if (Operation.equals("-")) {
-        type = 2;
-      }
-      else if (Operation.equals("*")) {
-        type = 3;
-      }
-      else if (Operation.equals("/")) {
-        type = 4;
-      }
-      else{
-        System.out.println("invalid operation");
-        return "stop";
-      }
+
+
+
+
 //Checkpoint 2 break down each number into charcter(s)
-String W1,N1,D1,W2,N2,D2;
+System.out.println( num1.length());
+System.out.println( num2.length());
+      String W1,N1,D1,W2,N2,D2;
       if (num1.indexOf("_") >= 0 ){//for have fraction
          W1 = num1.substring(0,num1.indexOf("_"));
          N1 = num1.substring(num1.indexOf("_")+1,num1.indexOf("/"));
-         D1 = num1.substring(num1.indexOf("/")+1);
+         D1 = num1.substring(num1.indexOf("/")+1,num1.length());
         System.out.println( "whole 1: "+W1);
         System.out.println("numerator 1: "+N1);
         System.out.println("denominator 1: "+D1);
-
+      }
+      else if (num1.contains("/")== true) {
```

Comment:
`.contains()` returns `true` or `false` so you don't need `== true`...the method evaluates to a boolean value already!